Ignore mantid packages in dependency spotter

jclarkeSTFC opened the pull request:
No need to output changes in the various mantid packages when running dependency_spotter.py

Small addition to #35866 

**Description of work**

Ignore packages whose names start with "mantid" when comparing package version changes between builds.

**To test:**

<!-- Instructions for testing.
There should be sufficient instructions for someone unfamiliar with the application to test - unless a specific
reviewer is requested.
If instructions for replicating the fault are contained in the linked issue then it is OK to refer back to these.
-->

*This does not require release notes* because **it's for a developer tool**

<!-- Ensure the base of this PR is correct (e.g. release-next or main)
Finally, don't forget to add the appropriate labels, milestones, etc.!  -->

---

#### Reviewer ####

Please comment on the points listed below ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)).
**Your comments will be used as part of the gatekeeper process, so please comment clearly on what you have checked during your review.** If changes are made to the PR during the review process then your final comment will be the most important for gatekeepers. In this comment you should make it clear why any earlier review is still valid, or confirm that all requested changes have been addressed.

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?
- Are the release notes saved in a separate file, using Issue or PR number for file name and in the correct location?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

#### Gatekeeper ####

If you need to request changes to a PR then please add a comment and set the review status to "Request changes". This will stop the PR from showing up in the list for other gatekeepers.

jclarkeSTFC committed:
Ignore mantid packages.

No need to output changes in the various mantid packages when running dependency_spotter.py

thomashampson reviewed (changes requested):
I tried running with these parameters and it gave me a KeyError:
`python tools/Jenkins/dependency_spotter.py -f 598 -s 605`
Can you look into that?

Also, I think it would be nice if you added `required=True` for the two parameters without defaults (`first` and `second`). Then you get a usage message printed out when you just run `python tools/Jenkins/dependency_spotter.py`.

jclarkeSTFC committed:
Fix bug when package added/removed.

Previously the tool was incorrectly checking for whether a package was in the dictionary or not.

jclarkeSTFC committed:
Fix bug when adding package names to list.

Use append() instead of extend() so you don't get every character in the string treated as a separate package.

jclarkeSTFC committed:
Print out URLs the script is using.

thomashampson reviewed (approved):
This removes the unwanted `mantid` package version comparison, and fixes a couple of bugs too. :shipit:
